boards/rp-pico/examples/pico_display_pack_2.rs

```diff
@@ -0,0 +1,170 @@
+
+// SPI display interface
+use display_interface_spi::SPIInterface;
+
+// Graphics drawing utilities
+use embedded_graphics::pixelcolor::Rgb565;
+use embedded_graphics::prelude::*;
+use embedded_graphics::primitives::*;
+
+// GPIO traits
+use embedded_hal::digital::v2::OutputPin;
+use embedded_hal::PwmPin;
+
+/// External high-speed crystal on the Raspberry Pi Pico board is 12 MHz. Adjust
+/// if your board has a different frequency
+const XTAL_FREQ_HZ: u32 = 12_000_000u32;
+
+#[entry]
+fn main() -> ! {
+    // Grab our singleton objects
+    let mut pac = pac::Peripherals::take().unwrap();
+    let core = pac::CorePeripherals::take().unwrap();
+
+    // Set up the watchdog driver - needed by the clock setup code
+    let mut watchdog = hal::Watchdog::new(pac.WATCHDOG);
+
+    // Configure the clocks
+    let clocks = hal::clocks::init_clocks_and_plls(
+        XTAL_FREQ_HZ,
+        pac.XOSC,
+        pac.CLOCKS,
+        pac.PLL_SYS,
+        pac.PLL_USB,
+        &mut pac.RESETS,
+        &mut watchdog,
+    )
+    .ok()
+    .unwrap();
+
+    let mut delay = cortex_m::delay::Delay::new(core.SYST, clocks.system_clock.freq().to_Hz());
+
+    // The single-cycle I/O block controls our GPIO pins
+    let sio = hal::Sio::new(pac.SIO);
+
+    // Set the pins up according to their function on this particular board
+    let pins = rp_pico::Pins::new(
+        pac.IO_BANK0,
+        pac.PADS_BANK0,
+        sio.gpio_bank0,
+        &mut pac.RESETS,
+    );
+
+    // Init PWMs
+    let mut pwm_slices = hal::pwm::Slices::new(pac.PWM, &mut pac.RESETS);
+
+    // Configure and enable PWMs
+    let pwm3 = &mut pwm_slices.pwm3;
+    pwm3.set_ph_correct();
+    pwm3.enable();
+    let pwm4 = &mut pwm_slices.pwm4;
+    pwm4.set_ph_correct();
+    pwm4.enable();
+
+    // Configure button inputs
+    // 12 - A button
+    let _a_pin = pins.gpio12.into_pull_up_input();
+    // 13 - B button
+    let _b_pin = pins.gpio13.into_pull_up_input();
+    // 14 - X button
+    let _x_pin = pins.gpio14.into_pull_up_input();
+    // 15 - Y button
+    let _y_pin = pins.gpio15.into_pull_up_input();
+
+    // 6 (PWM 3A) - LED red
+    let r_channel = &mut pwm3.channel_a;
+    r_channel.output_to(pins.gpio6);
+    r_channel.set_inverted();
+    r_channel.set_duty(0);
+    // 7 (PWM 3B) - LED green
+    let g_channel = &mut pwm3.channel_b;
+    g_channel.output_to(pins.gpio7);
+    g_channel.set_inverted();
+    g_channel.set_duty(0);
+    // 8 (PWM 4A) - LED blue
+    let b_channel = &mut pwm4.channel_a;
+    b_channel.output_to(pins.gpio8);
+    b_channel.set_inverted();
+    b_channel.set_duty(0);
+
+    // 20 - LCD backlight
+    pins.gpio20.into_push_pull_output().set_high().unwrap();
+
+    // 22 - LCD hard reset
+    let mut rst_pin = pins.gpio22.into_push_pull_output();
+    rst_pin.set_low().unwrap();
+
+    // Setup SPI
+    let spi_cs = pins.gpio17.into_push_pull_output();
+    let spi_miso = pins.gpio16.into_push_pull_output();
```

Comment:
This pin is not used as miso so I'd suggest giving it an name matching its use. eg: `di_data_command`